alg/sha256_sse2.c

```diff
@@ -225,10 +222,14 @@ SHA256_Transform_sse2(uint32_t state[static restrict 8],
 
 		if (i == 48)
 			break;
-		MSG4(W, 0, i);
-		MSG4(W, 4, i);
-		MSG4(W, 8, i);
-		MSG4(W, 12, i);
+		Y[0] = MSG4(Y[0], Y[1], Y[2], Y[3]);
+		_mm_storeu_si128((__m128i *)&W[16 + i + 0], Y[0]);
```

Comment:
Again, for performance reasons I'm interleaving MSG4 with _mm_storeu_si128().

We could avoid this by passing `(W, 0, i)` to `MSG4`, but IMO that looks uglier than this.  Another idea I've tried is to pass `(&W[16 + i + 0])` to `MSG4`... but again, that spoiled the "niceness" of MSG4 simply taking X0, X1, X2, X3.